pagopa/io-app
# feat: [IOAPPCOM-3] Add consent screen to manage push notifications

# ⚠️ This pr depends on #4142 ⚠️
Please review first #4142 in order to avoid duplicate review work.

## Short description
This PR, only for iOS, move the permission request to use notifications after opt-in screen.

**New User:**

https://user-images.githubusercontent.com/29163287/199494262-941056c7-58ca-455e-a968-60024f8fb557.mp4

**Existing User:**

https://user-images.githubusercontent.com/29163287/199491286-a92164ef-8864-4e2e-a818-9fce5934f8cb.mp4

## List of changes proposed in this pull request
- Added route `ONBOARDING_NOTIFICATIONS_INFO_SCREEN_CONSENT`
- Moved `updateInstallationSaga` in startup
- Added `checkNotificationsPermissionsSaga`
- Updated `checkNotificationsPreferencesSaga`
- Added screen `OnboardingNotificationsInfoScreenConsent`

## How to test
- Set `firstOnboarding` to `true` or `false` in the [dev server config](https://github.com/pagopa/io-dev-api-server/blob/02ea784cdf244759c9dd90cac42b60ad03491b58/src/config.ts#L63)
- Start dev server
- Perform a login
- Browse to the opt-in screen
- Existing User: 
    - Check permission
    - If you granted request, go ahead, otherwise go to the consent screen
- New user:
    - Request permission
    - If you grant request, go ahead, otherwise go to the consent screen
